Fixes

JuiP opened the pull request:
WIP

- [x] Port to python3

- [x] Clicking on the level drop down, no other levels are shown when hard level is selected

- [x] Clicking on the area beside 3,6,9 number buttons : mouse clicks enters number 4 and 7 in the dash

- [x] Clicking on the toolbar and then the add to desciption button causes black screen

- [x] Hard level dashes do not fit the screen

- [x] The smile is not centered

Fixes #16.

ayushnawal committed:
update sugargame to v1.3

ayushnawal committed:
matching semantics for sugargame v1.3
remove stop accelerator
add exit test
detect gtk stop button

ayushnawal committed:
Remove black screen when play button pressed

ayushnawal committed:
Fix force redraw on pressing play and next round button

ayushnawal committed:
Add icon for restart button

ayushnawal committed:
manage frame rate on screen redraw

ayushnawal committed:
handle videoresize

removed show_all to avoid any race condition

JuiP committed:
2to3 and other port to python3 changes

JuiP committed:
Fix Hard level number entry to fit the screen

JuiP committed:
Fix : Clicking on the level drop down and select hard, no other levels are seen

JuiP committed:
Force redraw

- Fixes bug: Click on the toolbar and then click add description. This causes black screen. Keep the mouse on the toolbar, once the mouse is moved over the canvas, black screen disappears

srevinsaju commented on CowBulls.py:
```diff
@@ -229,13 +240,12 @@ def run(self):
                 g.redraw = False
 
             pygame.display.update()
+            self.clock.tick(20) # Try to stay at 20 FPS
```
flake: E261 at least two spaces before inline comment

JuiP replied:
Done in b616cef

srevinsaju commented on activity.py:
```diff
@@ -42,19 +42,22 @@ def __init__(self, handle):
         # Build the activity toolbar.
         toolbox = ToolbarBox()
-
+        
```
flake: Trailing space

JuiP replied:
Done in b616cef

srevinsaju commented on g.py:
```diff
@@ -21,9 +21,9 @@ def init():
     pygame.display.flip()
     w, h = screen.get_size()
-    scale = min([w / 1200.0, h / 700.0])
+    scale = min([w / 1200.0, h / 900.0])
```
My suggestion, it would be great if `tuple` is used instead of `list`; not necessayy however

JuiP replied:
I don't think this is a necessary change, why do you think `tuple` should be used?

srevinsaju commented on icons/new-game.svg:
```diff
@@ -0,0 +1,91 @@
+    <path
+   d="m 42.645186,31.124796 c -1.10877,0 -2.018763,0.909992 -2.018763,2.018761 l 0,4.629689 -3.283849,-3.283849 c -0.784855,-0.783184 -2.043082,-0.78485 -2.826264,0 -0.783183,0.784854 -0.784823,2.069998 0,2.85318 l 3.310766,3.28385 -4.683521,0 c -1.10877,0 -2.018762,0.909991 -2.018762,2.018762 0,1.108769 0.909992,1.991843 2.018762,1.991843 l 4.656605,0 -3.28385,3.283849 c -0.783181,0.784854 -0.78 ...
+   id="rect3894"
+   style="fill:#ffffff;fill-opacity:1;stroke:#282828;stroke-width:1.72267997;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:4;stroke-opacity:1;stroke-dasharray:none" /></svg>
```
Missing newline at end of file

JuiP replied:
Done in 96e5e4b

srevinsaju commented on utils.py:
```diff
@@ -47,9 +47,9 @@ def get_num(level):
     while num[0] == 0:
         if level == 3:
-                num = random.sample(range(0, 10), 3)
+                num = random.sample(list(range(0, 10)), 3)
```
```python
>>> random.sample(range(0,10), 3)
[5, 0, 7]
>>> random.sample(range(0,10), 3)
[3, 9, 6]
>>> random.sample(range(0,10), 3)
[5, 6, 9]
>>> 
```
As you can see, there is no need of the `list` function, As mentioned [here](https://github.com/sugarlabs/sugar-docs/blob/master/src/python-porting-guide.md),
> Port from Python 2 to Python 3. Start your porting with 2to3 tool,
> In the terminal, type:
> `2to3 -w -n *.py`
> and then **_review_ every change made,**

JuiP replied:
Thanks for noticing! Done in 96e5e4b

srevinsaju commented on utils.py:
```diff
@@ -47,9 +47,9 @@ def get_num(level):
+                num = random.sample(list(range(0, 10)), 3)
         else:
-            num = [random.choice(range(0, 10)) for x in range(level)]
+            num = [random.choice(list(range(0, 10))) for x in range(level)]
```
This could be better written as
```python
num = [random.randint(0, 10) for _ in range(level)]
```
Additionally, I have also replaced `x` with `_`, so that we can avoid `F401` as part of `flake8`

JuiP replied:
Done in 96e5e4b

srevinsaju reviewed (changes requested):
Looks good. Did a code review; will test this soon :smile:
very descriptive commit messages :+1:

JuiP committed:
Flake8 changes

JuiP committed:
Changes

- Remove unnecessary changes made by 2to3 tool and change choice to randint for better code
- Fix EOL for new-game.svg
- Fix: flake8 last changes

JuiP committed:
Fix: The won/lost emoji is not centered

JuiP committed:
Fix TypeError
TypeError: randint() missing 1 required positional argument: 'b'
The Type Error also causes IndexError: list index out of range on line utils.blit_offset('status/' + self.compare_list[num][x],

To Reproduce the error: Input one numberwhen on Level `Easy` Then shift the level to `Hard`. The Error occurs

JuiP committed:
Fix Port to python3 regression and logical error

- To reproduce: Clicking on area beside 3,6,9 dialpad keys, mouse click enters number 4 and 7 in the input number dash

JuiP commented:
Tested; worked like expected.
@srevinsaju @chimosky @quozl Please test, review and merge :)

Merging this PR also closes #16

chimosky commented:
Reviewed, tested. Thanks.

*edited opening comment.*

The mouse issue you noticed still persists, have you found the cause or any speculations?
I'm yet to take a look.

JuiP commented:
Thanks! :)

Regarding the mouse issue, seems like it happens in every game that uses pygame, @quozl confirmed in [comment](https://github.com/sugarlabs/jumble-activity/pull/10#discussion_r447476763)

chimosky commented:
Thanks for that.
